sync-core/src/test/java/com/cloudant/sync/query/IndexCreatorTest.java

```diff
@@ -26,144 +32,110 @@
     }
 
     @Test
     public void preconditionsToCreatingIndexes() {
         // doesn't create an index on null fields
         String name = im.ensureIndexed(null, "basic");
-        Assert.assertNull(name);
+        assertThat(name, is(nullValue()));
 
         // doesn't create an index on no fields
         List<Object> fieldNames = new ArrayList<Object>();
         name = im.ensureIndexed(fieldNames, "basic");
-        Assert.assertNull(name);
+        assertThat(name, is(nullValue()));
 
         // doesn't create an index on null name
         fieldNames = Arrays.<Object>asList("name");
         name = im.ensureIndexed(fieldNames, null);
-        Assert.assertNull(name);
+        assertThat(name, is(nullValue()));
 
         // doesn't create an index without a name
         name = im.ensureIndexed(fieldNames, "");
-        Assert.assertNull(name);
+        assertThat(name, is(nullValue()));
 
         // doesn't create an index on null index type
         name = im.ensureIndexed(fieldNames, "basic", null);
-        Assert.assertNull(name);
+        assertThat(name, is(nullValue()));
 
         // doesn't create an index if duplicate fields
         fieldNames = Arrays.<Object>asList("age", "pet", "age");
         name = im.ensureIndexed(fieldNames, "basic");
-        Assert.assertNull(name);
+        assertThat(name, is(nullValue()));
     }
 
     @Test
     public void createIndexOverOneField() {
-        List<Object> fieldNames = Arrays.<Object>asList("name");
-        String name = im.ensureIndexed(fieldNames, "basic");
-        Assert.assertTrue(name.equals("basic"));
+        String indexName = im.ensureIndexed(Arrays.<Object>asList("name"), "basic");
+        assertThat(indexName, is(notNullValue()));
+        assertThat(indexName, is("basic"));
 
         Map<String, Object> indexes = im.listIndexes();
-        Assert.assertEquals(1, indexes.size());
-        Assert.assertTrue(indexes.containsKey("basic"));
+        assertThat(indexes, hasKey("basic"));
 
         @SuppressWarnings("unchecked")
         Map<String, Object> index = (Map<String, Object>) indexes.get("basic");
         @SuppressWarnings("unchecked")
         List<String> fields = (List<String>) index.get("fields");
-        Assert.assertEquals(3, fields.size());
-        Assert.assertTrue(fields.contains("_id"));
-        Assert.assertTrue(fields.contains("_rev"));
-        Assert.assertTrue(fields.contains("name"));
+        assertThat(fields, containsInAnyOrder("_id", "_rev", "name"));
     }
 
     @Test
     public void createIndexOverTwoFields() {
-        List<Object> fieldNames = Arrays.<Object>asList("name", "age");
-        String name = im.ensureIndexed(fieldNames, "basic");
-        Assert.assertTrue(name.equals("basic"));
+        String indexName = im.ensureIndexed(Arrays.<Object>asList("name", "age"), "basic");
+        assertThat(indexName, is(notNullValue()));
+        assertThat(indexName, is("basic"));
```

Comment:
Again, and throughout, `assertThat(indexName, is("basic"))` implies `assertThat(indexName, is(notNullValue()))`. Delete the null checks.